authors/apps/articles/views.py

```diff
@@ -0,0 +1,113 @@
+from rest_framework.generics import (
+    ListCreateAPIView, RetrieveUpdateDestroyAPIView)
+from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.exceptions import ValidationError
+
+from .models import Article
+from .serializers import ArticleSerializers
+from .renderers import ArticleJsonRenderer
+from authors.apps.core.permissions import IsOwnerOrReadonly
+
+
+def article_not_found():
+    raise ValidationError(
+        detail={'error': 'No article found for the slug given'})
+
+
+def get_article(slug):
+    """
+    Returns article with the given slug if exists
+    or returns an exception if no article with slug exists
+    """
+    try:
+        article = Article.objects.get(slug=slug)
+        return article
+    except Exception:
+        article_not_found()
+        return Response(status=status.HTTP_404_NOT_FOUND)
+
+
+class ListCreateArticle(ListCreateAPIView):
```

Comment:
Why have you used ListCreateAPIView? 
Remember to attach some description to define what ListCreateArticle class does.

Reply:
Hey @jomasim,
ListCreateAPIView provides get and post method handlers. It allows us to post articles and also gives us a get method which allows us to get a list of all the articles. It extends the ListModelMixin which provides a .list(request, *args, **kwargs) method, that implements listing a queryset. If the queryset is populated, this returns a 200 OK response, with a serialized representation of the queryset as the body of the response. The response data may optionally be paginated. And this will help when we come to the task of pagination.